src/main/scala/io/delta/execution/DeltaTableOperations.scala

```diff
@@ -68,4 +74,159 @@ trait DeltaTableOperations { self: DeltaTable =>
   }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * @param set
+   */
+  def update(set: Map[String, Column]): Unit = {
+    val setColumns = getSetColumnsForColumn(set)
+    executeUpdate(setColumns, None)
+  }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param set
+   */
+  def update(set: java.util.Map[String, Column]): Unit = {
+    val setColumns = getSetColumnsForJavaColumn(set)
+    executeUpdate(setColumns, None)
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * @param condition
+   * @param set
+   */
+  def update(condition: Column, set: Map[String, Column]): Unit = {
+    val setColumns = getSetColumnsForColumn(set)
+    executeUpdate(setColumns, Some(condition))
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param condition
+   * @param set
+   */
+  def update(condition: Column, set: java.util.Map[String, Column]): Unit = {
+    val setColumns = getSetColumnsForJavaColumn(set)
+    executeUpdate(setColumns, Some(condition))
+  }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * @param set
+   */
+  def updateExpr(set: Map[String, String]): Unit = {
+    val setColumns = getSetColumns(set)
+    executeUpdate(setColumns, None)
+  }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param set
+   */
+  def updateExpr(set: java.util.Map[String, String]): Unit = {
+    val setColumns = getSetColumnsForJava(set)
+    executeUpdate(setColumns, None)
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * @param condition
+   * @param set
+   */
+  def updateExpr(condition: String, set: Map[String, String]): Unit = {
+    val setColumns = getSetColumns(set)
+    executeUpdate(setColumns, Some(functions.expr(condition)))
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param condition
+   * @param set
+   */
+  def updateExpr(condition: String, set: java.util.Map[String, String]): Unit = {
+    val setColumns = getSetColumnsForJava(set)
+    executeUpdate(setColumns, Some(functions.expr(condition)))
+  }
+
+  protected def getSetColumns(set: Map[String, String]): Seq[(String, Column)] = {
+    var setColumns: Seq[(String, Column)] = Seq()
+    for ((col, expr) <- set) {
+      setColumns = setColumns :+ ((col, functions.expr(expr)))
+    }
+    setColumns
+  }
+
+  protected def getSetColumnsForColumn(set: Map[String, Column]): Seq[(String, Column)] = {
```

Comment:
these methods can be removed if 
- executeUpdate directly takes a scala map
- java map to scala map conversion is done with JavaConverters and `asScala`

Reply:
done.